File: modules/text-messager/src/main/java/com/gmail/marcosav2010/tm/PacketMessage.java
package com.gmail.marcosav2010.tm;

import com.gmail.marcosav2010.communicator.packet.Packet;
import com.gmail.marcosav2010.communicator.packet.wrapper.PacketDecoder;
import com.gmail.marcosav2010.communicator.packet.wrapper.PacketEncoder;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;

import java.io.IOException;

@AllArgsConstructor
@NoArgsConstructor
public class PacketMessage extends Packet {

    @Getter
    private String message;

    @Override
    protected void encodeContent(PacketEncoder out) throws IOException {
        out.write(message);
    }

    @Override
    protected void decodeContent(PacketDecoder in) throws IOException {
        message = in.readString();
    }
}

File: modules/text-messager/src/main/java/com/gmail/marcosav2010/tm/PacketMessage.kt
package com.gmail.marcosav2010.tm

import com.gmail.marcosav2010.communicator.packet.Packet
import com.gmail.marcosav2010.communicator.packet.wrapper.PacketDecoder
import com.gmail.marcosav2010.communicator.packet.wrapper.PacketEncoder
import kotlin.properties.Delegates

class PacketMessage() : Packet() {

    var message by Delegates.notNull<String>()
        private set

    constructor(message: String) : this() {
        this.message = message
    }

    override fun encodeContent(out: PacketEncoder) {
        out.write(message)
    }

    override fun decodeContent(`in`: PacketDecoder) {
        message = `in`.readString()
    }
}